include/substrait-mlir/Dialect/Substrait/IR/SubstraitOps.td

```diff
@@ -415,6 +415,25 @@ def Substrait_EmitOp : Substrait_RelOp<"emit", [
   }];
 }
 
+def Substrait_FetchOp : Substrait_RelOp<"fetch", [
+    DeclareOpInterfaceMethods<InferTypeOpInterface>
+  ]> {
+  let summary = "Fetch operation";
+  let description = [{
+    Represents a `FetchOp` message together with the `RelCommon` and input
+    `Rel` message it contains. 
+  }];
+  let arguments = (ins
+    Substrait_Relation:$input,
+    I64Attr: $offset,
+    I64Attr: $count
+  );
+  let results = (outs Substrait_Relation:$result);
+  let assemblyFormat = [{
+    $input `,` $offset `,` $count attr-dict `:` type($input) `->` type($result)
```

Comment:
Bonus: input and output type are always the same. This means that we can omit the type of one of the two in the assembly format if we specify the equality of the types using one of the constraints mentioned [here](https://mlir.llvm.org/docs/DefiningDialects/Operations/#type-inference). (That would also allow us to remove the `inferReturnType` function, I believe.)

Reply:
I think we can be a bit more creative here. For example `fetch 5 offset 10 from %input : tuple<si32>`. WDYT?

Super extra bonus: it'd be great if we can skip the part of `offset` part if it is 0 and/or print the `count` as `all` if the value is `-1`. I am pretty sure the latter requires a custom printer/parser directive, so maybe skip that in this PR.

Reply:
Will do the super extra bonus for the count part in a different PR. 😄 Managed to do the offset part in this PR.